source/MaskManager.h

```diff
@@ -0,0 +1,49 @@
+/* MaskManager.h
+Copyright (c) 2021 by Jonathan Steck
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#ifndef MASK_MANAGER_H_
+#define MASK_MANAGER_H_
+
+#include <vector>
+#include <map>
+
+class Mask;
+class Sprite;
+
+
+
+// Class that stores the masks for sprites that have them, and provides the correct
+// mask for the scale that the sprite requests.
+class MaskManager {
+public:
+	// Move the given masks at 1x scale into the manager's storage.
+	// The given vector will be cleared.
+	void AddMasks(const Sprite *sprite, std::vector<Mask> &masks);
+	
+	// Add a scale that the given sprite needs to have a mask for.
+	void AddScale(const Sprite *sprite, double scale);
+	
+	// Create the scaled versions of all masks from the 1x versions. 
+	void ScaleMasks();
+	
+	// Get the masks for the given sprite at the given scale. If a
+	// sprite has no masks, an empty mask is returned.
+	const std::vector<Mask> &GetMasks(const Sprite *sprite, double scale) const;
+	
+	
+private:
+	std::map<const Sprite *, std::map<double, std::vector<Mask>>> spriteMasks;
```

Comment:
One possible point of failure of this approach is that `float` and `double` can be terrible map keys, especially when they are derived from computations. In our case, we don't perform any computations (beyond `static_cast<double>`) on the `Body::scale` field, so we should effectively always be searching for the original value whenever we are looking for a scaled body's collision masks.

If collision mask retrieval does end up being an issue, we could enforce an arbitrary set of allowed scaling factors based on what enables good scaling (i.e. multiples of 2). We'd then convert the users' input "scale" into a power of 2 and then use that integer power as the map key.